packages/commerce-sdk-react/src/hooks/ShopperCustomers/mutation.ts

```diff
@@ -441,22 +439,49 @@ export const SHOPPER_CUSTOMERS_NOT_IMPLEMENTED = [
     'updateCustomerProductList'
 ]
 
+export type ShopperCustomersMutationType = typeof ShopperCustomersMutations[keyof typeof ShopperCustomersMutations]
+
+type UseShopperCustomersMutationHeaders = NonNullable<
+    Argument<Client['registerCustomer']>
+>['headers']
+type UseShopperCustomersMutationArg = {
+    headers?: UseShopperCustomersMutationHeaders
+    rawResponse?: boolean
+    action: ShopperCustomersMutationType
+}
+
+type ShopperCustomersClient = ApiClients['shopperCustomers']
+
 /**
  * A hook for performing mutations with the Shopper Customers API.
  */
-export function useShopperCustomersMutation<Action extends ShopperCustomersMutationType>(
-    action: Action
-) {
+function useShopperCustomersMutation<Action extends ShopperCustomersMutationType>(
+    arg: UseShopperCustomersMutationArg
+): UseMutationResult<
+    DataType<ShopperCustomersClient[Action]> | Response,
+    Error,
+    Argument<ShopperCustomersClient[Action]>
+> {
+    const {headers, rawResponse, action} = arg
+
     if (SHOPPER_CUSTOMERS_NOT_IMPLEMENTED.includes(action)) {
         NotImplementedError()
     }
-    type Params = Argument<Client[Action]>
-    type Data = DataType<Client[Action]>
+    type Params = Argument<ShopperCustomersClient[Action]>
+    type Data = DataType<ShopperCustomersClient[Action]>
     const queryClient = useQueryClient()
     return useMutation<Data, Error, Params>(
         (params, apiClients) => {
             const method = apiClients['shopperCustomers'][action] as MutationFunction<Data, Params>
-            return method.call(apiClients['shopperCustomers'], params)
+            //Override param headers to user-defined header values
+            if (params) {
+                params.headers = headers
+            }
+            return (method.call as (
+                apiClient: ShopperCustomersClient,
+                params: Params,
+                rawResponse: boolean | undefined
+            ) => any)(apiClients['shopperCustomers'], params, rawResponse)
```

Comment:
Overriding the `params` like this can introduce subtle bugs because the params still exist outside of the current scope. In general, whenever we get an object as input, the best practice is to avoid changing it directly, and we make a copy instead.

So in our case, we can create a new object with object destructuring like this:

```suggestion
            return (method.call as (
                apiClient: ShopperCustomersClient,
                params: Params,
                rawResponse: boolean | undefined
            ) => any)(apiClients['shopperCustomers'], {...params, headers}, rawResponse)
```